ev3dev2/motor.py

```diff
@@ -879,67 +898,67 @@ def _set_brake(self, brake):
         else:
             self.stop_action = self.STOP_ACTION_COAST
 
-    def on_for_rotations(self, speed_pct, rotations, brake=True, block=True):
+    def on_for_rotations(self, speed, rotations, brake=True, block=True):
         """
-        Rotate the motor at ``speed_pct`` for ``rotations``
+        Rotate the motor at ``speed`` for ``rotations``
 
-        ``speed_pct`` can be an integer percentage or a :class:`ev3dev2.motor.SpeedInteger`
+        ``speed`` can be a percentage or a :class:`ev3dev2.motor.SpeedMeasure`
         object, enabling use of other units.
         """
-        speed_pct = self._speed_pct(speed_pct)
+        speed = self._speed_native_units(speed)
 
-        if not speed_pct or not rotations:
-            log.warning("({}) Either speed_pct ({}) or rotations ({}) is invalid, motor will not move" .format(self, speed_pct, rotations))
+        if not speed or not rotations:
+            log.warning("({}) Either speed ({}) or rotations ({}) is invalid, motor will not move" .format(self, speed, rotations))
             self._set_brake(brake)
             return
 
-        self.speed_sp = int((speed_pct * self.max_speed) / 100)
-        self._set_position_rotations(speed_pct, rotations)
+        self.speed_sp = int(speed)
```

Comment:
How does `int(speed)` work? I don't see `__int__(self)` in definition of `SpeedMeasure`.

Reply:
Ah, I see that `speed` is already in native counts.

Reply:
The `_speed_native_units` call will convert the parameter to a number in counts/sec, assuming percentage if it's a straight integer. So here `speed` is just a number already.

This isn't exactly right, though; it should be `int(round(speed))` so we round rather than truncating.

Reply:
Yeah, the `Move*` classes convert to native units then do math on it. Back when we used percentages instead, this led to rounding two or three times, which I'd say is significant. The tests were having to use comparison margins of something like +/-7. That would be less bad now that we're using native units so rounding has roughly 1/10 of the effect, but still I'd like to avoid intermediate rounding as much as possible.

It might make sense to add a utility method somewhere to do the rounding… not sure if it's worth it.

Reply:
Ok, that is a good enough explanation for me.